src/compiler/transformers/map-imports-to-path-aliases.ts

```diff
@@ -28,12 +41,24 @@ export const mapImportsToPathAliases = (config: d.Config): ts.TransformerFactory
         if (!importPath.startsWith('.')) {
           const module = ts.resolveModuleName(importPath, sourceFile, config.tsCompilerOptions, compilerHost);
 
-          if (
-            module.resolvedModule?.isExternalLibraryImport === false &&
-            module.resolvedModule?.resolvedFileName != null
-          ) {
+          const hasResolvedFileName = module.resolvedModule?.resolvedFileName != null;
+          const isModuleFromNodeModules = module.resolvedModule?.isExternalLibraryImport === true;
+          const shouldTranspileImportPath = hasResolvedFileName && !isModuleFromNodeModules;
+
+          if (shouldTranspileImportPath) {
+            // Create a global extension matching regular expression
+            // This can be used to strip all file extensions off the generated import path string
+            const extensionRegex = new RegExp(Object.values(ts.Extension).join('|'), 'g');
```

Comment:
I think this is close, but still not quite right.  That's on me, I think my previous comment/concern wasn't worded well. In the initial implementation:

```ts
relative(sourceFile, module.resolvedModule.resolvedFileName).replace(module.resolvedModule.extension, '')
```

we would always replace the first instance of anything that matched the resolved module's extension. So if we had something like
```
foo.mjs.mjs
```
this would become
```
foo.mjs
```
which seemed correct, until we grouped the extensions:
```
foo[.mjs][.mjs]
// the first '.mjs' is what would get removed
foo[.mjs] // <- second extension stuck around
```


I think we still run into an issue here, as we might be removing more than we'd like for cases like
```
foo.mjs.js
```
we'd end up removing all the extensions, which I don't _think_ is what we want. Instead, we should rm the last entry only with something like:

```diff
- const extensionRegex = new RegExp(Object.values(ts.Extension).join('|'), 'g');
+ const extensionRegex = new RegExp(Object.values(ts.Extension).map((ext) => `${ext}$`).join('|'));
```

where:
- we append '$' to each extension to check that the filename ends with the extension
- we remove the 'g' flag, since we only want to remove one thing

Does that make sense? Or did I miss something here? 😆

Reply:
@rwaskiewicz This was kinda where I started, but then I tested a file with weird extensions like `foo.mts.ts`. This resulted in an import path like `../foo.mts` which throws an error because it's not a JS file import. Not sure how realistic files like that are, but it was causing issues

Reply:
Yeah, IDK of realistic it is, kind of a weird edge case for someone to have something like that working in Stencil 🤷

Reply:
@rwaskiewicz with that said, do we feel comfortable removing just the last extension, or want to play it safe and remove all?

Reply:
I say remove the last extension. If someone reports a bug, they can directly link this comment and blame me lol 😆